lib/decoder.js

```diff
@@ -813,7 +813,10 @@ var JpegImage = (function jpegImage() {
               offset--;
             }
             break;
-
+          case 0xE0:
+          case 0xE1: // Unused bytes
```

Comment:
what about moving this into `default` block and something like...

```js
// Recover from malformed APP1 markers popular in some phone models.
// See https://github.com/eugeneware/jpeg-js/issues/82
if (data[offset - 2] === 0x00 && data[offset - 1] === 0xE1) {
  const nextOffset = readUint16();
  if (data[nextOffset] === 0xFF) {
    offset = nextOffset;
    break;
  }
}
```

Reply:
Totally agree with you. In hindsight I should have seen that some of the files contained more than 1 `0xFFE1` segment header and concluded that they aren't unique. Headers should also always start with `0xFF` according to the JFIF standards, so you make a good point that `0x00E1` is definitely an error.

I've implemented the changes and added a new error that specifies the locations of 2 malformed headers like `0x00E1`. Let me know if there's anything else that can be improved!